### PR TITLE
added ability to specify options for vector and metadata indexes

### PR DESCRIPTION
**Pull request by epinzur**
related to https://github.com/CassioML/cassio/issues/36, but I'm wondering if my approach is too general. 

Should we just have an option to set `WITH OPTIONS = { 'similarity_function': 'dot_product' }` and not worry about the general use case?

Let me know your thoughts

### CODE REVIEW

**Comment by cbornet** [comment]
@epinzur Please check https://github.com/CassioML/cassio/pull/142 .
This will make available generic table options that are needed to support index analyzers.

**Comment by hemidactylus** [comment]
Hello, @epinzur , with #142 just merged it can be the right time to bring this one to completion. I can take care of most of it if you prefer - let's talk tomorrow, shall we? Cheers!

File: src/cassio/table/base_table.py [hunk 1]
```diff
@@ -407,32 +407,50 @@ def _get_db_setup_cql(self, schema: Dict[str, List[ColumnSpecType]]) -> str:
...
+            options_clause = f"WITH OPTIONS = {{{{ {options_text} }}}}"
+
+        return CREATE_INDEX_CQL_TEMPLATE.format(
+            index_name=index_name,
+            index_column=index_column,
+            options_clause=options_clause,
+        )
+
+    @staticmethod
+    def _get_create_analyzer_index_cql(index_options: List[Tuple[str, Any]]) -> str:
```
Comment (epinzur) [inline src/cassio/table/base_table.py:446]: moved most of the functionality from this method that @cbornet introduced into the `_get_create_index_cql()` method above, so I can share it with the vector index creation.

File: src/cassio/table/base_table.py [hunk 2]
```diff
@@ -407,32 +407,50 @@ def _get_db_setup_cql(self, schema: Dict[str, List[ColumnSpecType]]) -> str:
...
+                        formatted_options.append(f"'{key}': true")
+                    else:
+                        formatted_options.append(f"'{key}': false")
                 else:
-                    body_index_options.append(f"'{key}': false")
-            else:
-                raise ValueError("Unsupported body_index_option format")
+                    raise ValueError("Unsupported index_option format")
+
+            formatted_options.sort()
```
Comment (epinzur) [inline src/cassio/table/base_table.py:431]: sort options to limit test flakiness

File: src/cassio/table/base_table.py [hunk 3]
```diff
@@ -407,32 +407,50 @@ def _get_db_setup_cql(self, schema: Dict[str, List[ColumnSpecType]]) -> str:
         )
         return create_table_cql
 
-    def _get_create_analyzer_index_cql(
-        self, index_options: List[Tuple[str, Any]]
+    @staticmethod
+    def _get_create_index_cql(
```
Comment (epinzur) [inline src/cassio/table/base_table.py:411]: this method is static again (per comment #2)

File: src/cassio/table/cql.py [hunk 4]
```diff
@@ -26,12 +26,7 @@ class CQLOpType(Enum):
 
 CREATE_INDEX_CQL_TEMPLATE = (
     CREATE_INDEX_CQL_PREFIX
-    + "({index_column}) USING 'org.apache.cassandra.index.sai.StorageAttachedIndex';"
-)
-
-CREATE_INDEX_ANALYZER_CQL_TEMPLATE = (
-    CREATE_INDEX_CQL_PREFIX
-    + "({index_column}) USING 'org.apache.cassandra.index.sai.StorageAttachedIndex' WITH OPTIONS = {{{{{body_index_options}}}}};"  # noqa: E501
+    + "({index_column}) USING 'org.apache.cassandra.index.sai.StorageAttachedIndex' {options_clause};"  # noqa: E501
```
Comment (epinzur) [inline src/cassio/table/cql.py:29]: reverted to a single template for these 2

File: src/cassio/table/mixins.py [hunk 5]
```diff
@@ -581,6 +580,8 @@ def __init__(
         self,
         *pargs: Any,
         vector_dimension: Union[int, Awaitable[int]],
+        vector_similarity_function: Optional[str] = None,
+        vector_source_model: Optional[str] = None,
```
Comment (epinzur) [inline src/cassio/table/mixins.py:584]: from comment 1

**Comment by hemidactylus** [comment]
LGTM

**Pull request merged**